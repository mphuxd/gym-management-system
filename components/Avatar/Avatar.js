import React from 'react';
import * as AvatarPrimitive from '@radix-ui/react-avatar';
import styles from './Avatar.module.scss';

const Avatar = React.forwardRef(
  ({ children, id, src, alt, ...props }, forwardedRef) => (
    <AvatarPrimitive.Root
      className={styles.AvatarRoot}
      ref={forwardedRef}
      {...props}
    >
      <AvatarPrimitive.Image
        className="rounded-full h-12 w-12"
        src={src}
        alt={alt}
      />
    </AvatarPrimitive.Root>
  )
);

Avatar.displayName = 'Avatar';
export default Avatar;
