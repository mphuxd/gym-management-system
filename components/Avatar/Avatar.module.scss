@use "/styles/0-settings/sizes";
@use "/styles/0-settings/colors";

.AvatarRoot {
    display: inline-flex;
    align-items: center;
    justify-content: center;
    vertical-align: middle;
    overflow: hidden;
    user-select: none;
    width: 48px;
    height: 48px;
    border-radius: 100%;
}

.AvatarImage {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: inherit;
}

.AvatarFallback {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: white;
    color: colors.$gray-600;
    font-size: 15px;
}