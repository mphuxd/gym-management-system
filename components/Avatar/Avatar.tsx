import React from 'react';
import * as AvatarPrimitive from '@radix-ui/react-avatar';

export const AVATAR_TEST_ID = 'avatarTestId';

type AvatarRef = React.ElementRef<typeof AvatarPrimitive.Root>;
export interface AvatarProps extends AvatarPrimitive.AvatarProps {
  id?: string;
  src: string;
  alt: string;
}

const Avatar = React.forwardRef<AvatarRef, AvatarProps>(
  ({ children, id, src, alt, ...props }, forwardedRef) => (
    <AvatarPrimitive.Root
      id={id}
      className="inline-flex h-12 w-12 select-none items-center justify-center overflow-hidden rounded-full align-middle"
      ref={forwardedRef}
      {...props}
      data-testid={AVATAR_TEST_ID}
    >
      <AvatarPrimitive.Image
        className="h-12 w-12 rounded-full"
        src={src}
        alt={alt}
      />
    </AvatarPrimitive.Root>
  )
);

Avatar.displayName = 'Avatar';
export default Avatar;
