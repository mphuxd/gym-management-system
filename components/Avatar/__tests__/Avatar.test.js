import React from 'react';
import { render, screen } from '@testing-library/react';
import Avatar, { AVATAR_TEST_ID } from '../Avatar';

describe('Avatar ', () => {
  it('should match snapshot', async () => {
    const src = '/public/images/image-placeholder.png';
    const alt = 'placeholder alt';
    render(<Avatar src={src} alt={alt} />);

    const avatar = await screen.findByTestId(AVATAR_TEST_ID);
    expect(avatar).toMatchSnapshot();
  });
});
