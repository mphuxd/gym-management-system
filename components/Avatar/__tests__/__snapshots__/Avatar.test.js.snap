// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar  should match snapshot 1`] = `
<span
  class="inline-flex h-12 w-12 select-none items-center justify-center overflow-hidden rounded-full align-middle"
  data-testid="avatarTestId"
/>
`;
